pkg/upstream/cluster/loadbalancer.go

```diff
@@ -624,3 +625,95 @@ func (lb *reqRoundRobinLoadBalancer) IsExistsHosts(metadata api.MetadataMatchCri
 func (lb *reqRoundRobinLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) int {
 	return lb.hosts.Size()
 }
+
+type shortestResponseLoadBalancer struct {
+	hosts  types.HostSet
+	rand   *rand.Rand
+	mutex  sync.Mutex
+	choice uint32
+}
+
+func newShortestResponseLoadBalancer(_ types.ClusterInfo, hosts types.HostSet) types.LoadBalancer {
+	lb := &shortestResponseLoadBalancer{
+		hosts:  hosts,
+		rand:   rand.New(rand.NewSource(time.Now().UnixNano())),
+		choice: default_choice,
+	}
+	return lb
+}
+
+func (lb *shortestResponseLoadBalancer) ChooseHost(_ types.LoadBalancerContext) types.Host {
+	hs := lb.hosts
+	total := hs.Size()
+
+	if total == 0 {
+		return nil
+	}
+	if total == 1 {
+		return hs.Get(0)
+	}
+
+	lb.mutex.Lock()
+	defer lb.mutex.Unlock()
+
+	choice := total
+	rnd := false
+	// Choose `choice` times and return the best one.
+	// See The Power of Two Random Choices: A Survey of Techniques and Results
+	//  http://www.eecs.harvard.edu/~michaelm/postscripts/handbook2001.pdf
+	if choice > int(lb.choice) {
+		choice = int(lb.choice)
+		rnd = true
+	}
+
+	var candidates []types.Host
+	for i := 0; i < choice; i++ {
+		idx := i
+		if rnd {
+			idx = lb.rand.Intn(total)
+		}
+		temp := hs.Get(idx)
+
+		// Newly added host always has the shortest response
+		if temp.HostStats().UpstreamRequestTotal.Count() == 0 {
```

Comment:
Metrics can be disabled by config.
if host metrics are disabled , this load balancer cannot use.
maybe should check it ?

Reply:
Yes, this load balancer must working with metrics. And I didn't find where could disable the metrics, is there any document?

Reply:
Hello, @nejisama . I didn't find the config which to disable the metrics. The only way that could make an impact is `feedback` when shm is not configured.

https://github.com/mosn/mosn/blob/master/pkg/metrics/shm/configs.go#L20-L21

Reply:
Added logic to check and downgrade to WRR if disabled